tchannel/tornado/dispatch.py

```diff
@@ -182,9 +184,10 @@ def handle_call(self, request, connection):
                 )
 
                 # get new top-level resp from controller
-                new_resp = yield gen.maybe_future(
-                    handler.endpoint(new_req)
-                )
+                with StackContext(lambda: RequestContext(request.tracing)):
+                    f = handler.endpoint(new_req)
```

Comment:
Can we also move the StackContext management to the context module? As said previously, I would like for those details to not be leaked outside that module. This block of code should ideally look like this:

    with RequestContext(request.tracing):
        # ...

Or something similar.